oneflow/api/cpp/framework/graph.cpp

```diff
@@ -0,0 +1,267 @@
+  }
+  op_conf.set_device_tag(GetDeviceTag(device_));
+  if (batch_size_ > 0 && op_conf.has_input_conf()) {
+    op_conf.mutable_input_conf()->mutable_blob_conf()->mutable_shape()->mutable_dim()->Set(
+        0, batch_size_);
+    std::cout << "Print input conf" << std::endl;
+    std::cout << op_conf.ShortDebugString() << std::endl;
+  }
+  auto* ctx = JUST(of::GetCurInferCtx());
+  JUST(ctx->AddAndInferConsistentOp(op_conf));
+  return of::Maybe<void>::Ok();
+}
+
+of::Maybe<void> Graph::BuildGraph(const std::vector<Tensor>& inputs) {
+  CompileScope build_graph_scope(job_.job_conf(), *device_.device_->shared_from_symbol(), xrt_kind_);
+  {
+    // TODO(zzk0): remove this; used for input tensor order
+    int input_tensor_order = 0;
+    of::OpGraph op_graph(job_);
+    JUST(op_graph.ForEachOpNode([&](const of::OpNode& node) -> of::Maybe<void> {
+      const of::OperatorConf& op_conf = node.op().op_conf();
+      JUST(AddOp(op_conf));
+      if (op_conf.has_input_conf()) {
+        // TODO(zzk0): input tensor order
+        input_name_to_tensor_[op_conf.name()] = inputs.at(input_tensor_order++).tensor_;
+      } else if (op_conf.has_variable_conf()) {
+        // TODO(zzk0): load from local path, this branch maybe removed
+        of::LazyMode::Guard lazy_mode_disabled_guard{false};
+
+        of::VariableOpConf variable_conf = op_conf.variable_conf();
+        variable_op_name_to_tensor_[op_conf.name()] = JUST(of::one::functional::Rand(
+            of::Shape(variable_conf.shape()),
+            JUST(of::DType::Get(static_cast<of::DataType>(variable_conf.data_type()))),
+            *device_.device_, nullptr, false));
+        PrintTensor(Tensor(variable_op_name_to_tensor_[op_conf.name()]));
+      }
+      return of::Maybe<void>::Ok();
+    }));
+  }
+  JUST(of::CurJobBuildAndInferCtx_Complete());
+  JUST(of::CurJobBuildAndInferCtx_Rebuild());
+  {
+    std::shared_ptr<of::Job> complete_job = JUST(of::GetCurrentJob());
+    of::OpGraph complete_graph(*complete_job);
+    JUST(complete_graph.ForEachOpNode([&](const of::OpNode& node) -> of::Maybe<void> {
+      of::LazyMode::Guard lazy_mode_disabled_guard{false};
+      const of::OperatorConf& op_conf = node.op().op_conf();
+      if (op_conf.has_output_conf()) {
+        of::InterfaceBlobConf blob_conf = op_conf.output_conf().blob_conf();
+        output_name_to_tensor_[op_conf.name()] = JUST(of::one::functional::Empty(
+            of::Shape(blob_conf.shape()),
+            JUST(of::DType::Get(static_cast<of::DataType>(blob_conf.data_type()))),
+            *device_.device_));
+        std::cout << "Print output conf" << std::endl;
+        PrintTensor(Tensor(output_name_to_tensor_[op_conf.name()]));
+      }
+      return of::Maybe<void>::Ok();
+    }));
+  }
+  return of::Maybe<void>::Ok();
+}
+
+of::Maybe<void> Graph::LoadCheckpoint() { return of::Maybe<void>::Ok(); }
+
+of::Maybe<void> Graph::RegisterTensors() {
+  {
+    auto pair = Unzip(input_name_to_tensor_);
```

Comment:
```suggestion
    const auto pair = Unzip(input_name_to_tensor_);
```
能 const 就 const https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#con1-by-default-make-objects-immutable

Reply:
好的